### ISSUE TITLE
`applyVariables` with `property` and default widget produces erroneous problem report

### ISSUE THREAD

**Issue by rogerl50** [ArnoldSmith86/virtualtabletop#331]
The following `clickRoutine` code
```
    {
      "func": "COMPUTE",
      "operation": "concat",
      "operand1": "foo",
      "applyVariables": [
        {
          "parameter": "operand2",
          "property": "id"
        }
      ],
      "variable": "bar"
    },
```
will return the right thing ("foo" followed by the id of the invoking button), but the debug output erroneously says "Widget ID undefined does not exist". The message comes from isValidId, which assumes it really is looking for a valid widget ID.

**Comment by 96LawDawg** [comment]
Maybe I misunderstand, but isn't this a problem with your JSON?  How is it supposed to know what "property":"id" is supposed to be applied.  Maybe I'm going about it the long way and it should work your way, but I would get the id that I want and put it in a variable.  And then apply that variable.  I mean, the function is "applyVariables" and not "applyProperty."

**Comment by robartsd** [comment]
I think the idea was that the default widget would be the button that is running the routine if widget was left blank. Would be better if problem report gave  a warning that the value was missing, so the default was used.

**Comment by rogerl50** [comment]
> Maybe I misunderstand, but isn't this a problem with your JSON? How is it supposed to know what "property":"id" is supposed to be applied. Maybe I'm going about it the long way and it should work your way, but I would get the id that I want and put it in a variable. And then apply that variable. I mean, the function is "applyVariables" and not "applyProperty."

If the widget property is missing, it assumes that you are discussing the current widget (i.e. the invoking button)

**Comment by rogerl50** [comment]
> I think the idea was that the default widget would be the button that is running the routine if widget was left blank. Would be better if problem report gave a warning that the value was missing, so the default was used.

Yes, exactly. I was not as clear as I could have been in the problem report, but that is exactly what should happen. (And as I noted, the button running the routine is, in the end, substituted for the undefined ID and things work as they should).

**Comment by ArnoldSmith86** [comment]
Should be fixed with #333.
